search-app/src/cmr/search/api/routes.clj

```diff
@@ -231,14 +238,44 @@
         results (query-svc/find-concepts-by-parameters ctx concept-type search-params)]
     (search-response ctx results)))
 
+(defn- find-granule-parent-collections
+  "Invokes query service to find collections based on data found in a granule search,
+   then executes a search for collections with the found concept-ids. Supports CMR Harvesting."
+  [ctx path-w-extension params headers body]
+  (let [concept-type (concept-type-path-w-extension->concept-type path-w-extension)
+        ctx (assoc ctx :query-string body)
+        params (process-params params path-w-extension headers mt/xml)
+        collections-with-new-granules (query-svc/new-granules->collection-ids ctx params)
+        search-params (-> params
+                          (assoc :echo-collection-id collections-with-new-granules)
+                          (dissoc :has-granules-created-at)
+                          lp/process-legacy-psa)]
+    (find-concepts-by-parameters ctx path-w-extension search-params headers body)))
+
+(defn- multi-part-query?
+  "Return true if parameters match any from the list of `multi-part-query-params`
+   defined above. Supports CMR Harvesting."
+  [search-parameters]
```

Comment:
I tend to default to longer parameter names because autocomplete makes it trivial to type them, but I see your point. Fixed